File: Java/api/treehopper/src/main/java/io/treehopper/SoftPwmConfig.java
package io.treehopper;

/**
 * Internal SoftPWM config structure
 */
public class SoftPwmConfig {
    public SoftPwmConfig(Pin pin, int pulseWidthUs, boolean usePulseWidth)
    {
        this.Pin = pin;
        this.PulseWidthUs = pulseWidthUs;
        this.UsePulseWidth = usePulseWidth;
    }

    public Pin Pin;

    ///     Duty Cycle, from 0 to 1.
    public double DutyCycle;

    ///     Pulse Width, in Milliseconds
    public double PulseWidthUs;

    public boolean UsePulseWidth;

    public int Ticks;
}

File: NET/API/Treehopper/SoftPwmConfig.cs
namespace Treehopper
{
    internal class SoftPwmPinConfig
    {
        public Pin Pin { get; set; }

        /// <summary>
        ///     Duty Cycle, from 0 to 1.
        /// </summary>
        public double DutyCycle { get; set; }

        /// <summary>
        ///     Pulse Width, in Microseconds
        /// </summary>
        public double PulseWidthUs { get; set; }

        public bool UsePulseWidth { get; set; }

        public ushort Ticks { get; set; }
    }
}